File: app/src/main/java/com/apps/adrcotfas/goodtime/Util/VibrationPatterns.java
package com.apps.adrcotfas.goodtime.Util;

public class VibrationPatterns {

    private static final long[] DEFAULT_PATTERN = {
            0,
            250, 250, 250,
            250, 250, 250,
            250, 250, 250,
            250, 250, 250,
    };

    private static final int dot = 150;          // Length of a Morse Code "dot" in milliseconds
    private static final int dash = 375;         // Length of a Morse Code "dash" in milliseconds
    private static final int short_gap = 150;    // Length of Gap Between dots/dashes
    private static final int medium_gap = 375;   // Length of Gap Between Letters

    private static final long[] SOS_PATTERN = {
            0,
            dot, short_gap, dot, short_gap, dot,
            medium_gap,
            dash, short_gap, dash, short_gap, dash,
            medium_gap,
            dot, short_gap, dot, short_gap, dot
    };

    private static final int beat = 250;
    private static final int interbeat = 100;
    private static final int between_beat_pairs = 700;
    private static final long[] HEARTBEAT_PATTERN = {
            0,
            beat, interbeat, beat,
            between_beat_pairs,
            beat, interbeat, beat,
    };

    private static final long[] JACKHAMMER_PATTERN = {
            0,
            100, 100,
            100, 100,
            100, 100,
            100, 100,
            100, 100,
            100, 100,
            100, 100,
            100, 100,
            100, 100,
            100, 100,
            100, 100,
            100, 100,
            100
    };

    public static final long[][] LIST = {DEFAULT_PATTERN, SOS_PATTERN, HEARTBEAT_PATTERN, JACKHAMMER_PATTERN};
}

File: app/src/main/java/com/apps/adrcotfas/goodtime/util/VibrationPatterns.kt
package com.apps.adrcotfas.goodtime.util


object VibrationPatterns {
    private val NONE = longArrayOf()
    private val SOFT = longArrayOf(0, 50, 50, 50, 50, 50)
    private val STRONG = longArrayOf(0, 500, 500, 500)
    private const val DOT = 150
    private const val DASH = 375
    private const val SHORT_GAP = 150
    private const val MEDIUM_GAP = 375
    private val SOS_PATTERN = longArrayOf(
        0,
        DOT.toLong(), SHORT_GAP.toLong(), DOT.toLong(), SHORT_GAP.toLong(), DOT.toLong(),
        MEDIUM_GAP.toLong(),
        DASH.toLong(), SHORT_GAP.toLong(), DASH.toLong(), SHORT_GAP.toLong(), DASH.toLong(),
        MEDIUM_GAP.toLong(),
        DOT.toLong(), SHORT_GAP.toLong(), DOT.toLong(), SHORT_GAP.toLong(), DOT
            .toLong()
    )
    private const val BEAT = 250
    private const val INTER_BEAT = 100
    private const val BETWEEN_BEAT_PAIRS = 700
    private val HEARTBEAT_PATTERN = longArrayOf(
        0,
        BEAT.toLong(), INTER_BEAT.toLong(), BEAT.toLong(),
        BETWEEN_BEAT_PAIRS.toLong(),
        BEAT.toLong(), INTER_BEAT.toLong(), BEAT.toLong()
    )
    val LIST = arrayOf(NONE, SOFT, STRONG, SOS_PATTERN, HEARTBEAT_PATTERN)
}